aten/src/ATen/native/native_functions.yaml

```diff
@@ -1164,3 +1164,383 @@
+  dispatch:
+    SparseCPU: mul_sparse_scalar_
+
+- func: mul_out(Tensor result, Tensor self, Tensor other) -> Tensor
+  variants: function
+
+- func: mul_out(Tensor result, Tensor self, Scalar other) -> Tensor
+  variants: function
+
+- func: mul(Tensor self, Tensor other) -> Tensor
+  variants: method, function
+
+- func: mul(Tensor self, Scalar other) -> Tensor
+  variants: method, function
+
+- func: mul_(Tensor self, Tensor other) -> Tensor
+  variants: method
+
+- func: mul_(Tensor self, Scalar other) -> Tensor
+  variants: method
+
+
+
+- func: native_div_out(Tensor result, Tensor self, Scalar other) -> Tensor
+  variants: function
+  dispatch:
+    SparseCPU: div_out_sparse_scalar
+
+- func: native_div(Tensor self, Scalar other) -> Tensor
+  variants: function
+  dispatch:
+    SparseCPU: div_sparse_scalar
+
+- func: native_div_(Tensor self, Scalar other) -> Tensor
+  variants: function
+  dispatch:
+    SparseCPU: div_sparse_scalar_
+
+- func: div_out(Tensor result, Tensor self, Scalar other) -> Tensor
+  variants: function
+
+- func: div(Tensor self, Scalar other) -> Tensor
+  variants: method, function
+
+- func: div_(Tensor self, Scalar other) -> Tensor
+  variants: method
+
+
+- func: s_native_addmm_out(Tensor result, Tensor self, Tensor mat1, Tensor mat2, *, Scalar beta=1, Scalar alpha=1) -> Tensor
+  variants: function
+  dispatch:
+    CPU: s_addmm_out_sparse_dense_cpu
+
+- func: s_native_addmm(Tensor self, Tensor mat1, Tensor mat2, *, Scalar beta=1, Scalar alpha=1) -> Tensor
+  variants: function
+  dispatch:
+    CPU: s_addmm_sparse_dense_cpu
+
+- func: s_native_addmm_(Tensor self, Tensor mat1, Tensor mat2, *, Scalar beta=1, Scalar alpha=1) -> Tensor
+  variants: function
+  dispatch:
+    CPU: s_addmm_sparse_dense_cpu_
+
+- func: addmm_out(Tensor result, Tensor self, Tensor mat1, Tensor mat2, *, Scalar beta=1, Scalar alpha=1) -> Tensor
+  variants: function
+
+- func: addmm(Tensor self, Tensor mat1, Tensor mat2, *, Scalar beta=1, Scalar alpha=1) -> Tensor
+  variants: method, function
+
+- func: addmm_(Tensor self, Tensor mat1, Tensor mat2, *, Scalar beta=1, Scalar alpha=1) -> Tensor
+  variants: method
+
+
+- func: native_tensor(Type self_ty) -> Tensor
+  variants: function
+  dispatch:
+    SparseCPU: new_sparse
+
+- func: native_tensor(Type self_ty, IntList size) -> Tensor
+  variants: function
+  dispatch:
+    SparseCPU: new_with_size_sparse
+
+- func: tensor(Type dtype) -> Tensor
+  variants: function
+
+- func: tensor(Type dtype, IntList size) -> Tensor
+  variants: function
+
+
+# NB: The function overloads are removed to avoid a nasty bug where
+# you say at::native_sparse_coo_tensor(indices, values), and then
+# it does the dispatch based on indices (wrong wrong wrong!)  Without
+# the variants, you must call this on type directly.
+#
+# Maybe this meant we were supposed to take a dtype as an argument here.
+# Hmmmmm.
+
+- func: native_sparse_coo_tensor(IndexTensor indices, Tensor values) -> Tensor
+  variants: []
```

Comment:
wait -- what does this mean?

Reply:
`variants: []` means, don't generate `at::native_sparse_coo_tensor` or `Tensor:: native_sparse_coo_tensor`, but do create a method on `Type`. But I think I'm going to see if I can just make these take a `Type`.

Reply:
I think we should consider methods on types to be an implementation detail, it doesn't match up with the (non-legacy) pytorch API.